suricata/update/rule.py

```diff
@@ -118,7 +118,14 @@ def id(self):
         :returns: A tuple (gid, sid) representing the ID of the rule
         :rtype: A tuple of 2 ints
         """
-        return (int(self.gid), int(self.sid))
+        
+        self_gid=self.gid
+        self_sid=self.sid
+        if self.gid is not None:
+            self_gid=int(self_gid)
+        if self.sid is not None:
+            self_sid=int(self_sid)
+        return (self_gid, self_sid)
```

Comment:
This is incorrect. `sid` and `gid` fields are pretty improtant for any rule and are never set to `NoneType` after parsing. You should instead handle the exception cleanly so that in case of failure there is no ugly traceback.